src/modules/cpu/kernel/fmadd_scalar.hpp

```diff
@@ -0,0 +1,126 @@
+    {
+        RpptROI3D roi;
+        RpptROI3DPtr roiPtrInput = &roiGenericPtrSrc[batchCount];
+        compute_roi3D_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f mulParam = mulTensor[batchCount];
+        Rpp32f addParam = addTensor[batchCount];
+
+        Rpp32f *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrImage = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        Rpp32u bufferLength = roi.xyzwhdROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp32f *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xyzwhdROI.xyz.z * srcGenericDescPtr->strides[2]) + (roi.xyzwhdROI.xyz.y * srcGenericDescPtr->strides[3]) + (roi.xyzwhdROI.xyz.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+
+#if __AVX2__
+        Rpp32u vectorIncrementPerChannel = 8;
+
+        __m256 pFmaddParams[2];
+        pFmaddParams[0] = _mm256_set1_ps(mulParam);
+        pFmaddParams[1] = _mm256_set1_ps(addParam);
+#endif
+        // Fmadd without fused output-layout toggle (NDHWC -> NDHWC or NCDHW -> NCDHW)
+#if __AVX2__
+        Rpp32u alignedLength = bufferLength & ~(vectorIncrementPerChannel-1);
+#endif
+        for(int c = 0; c < layoutParams.channelParam; c++)
+        {
+            Rpp32f *srcPtrDepth, *dstPtrDepth;
+            srcPtrDepth = srcPtrChannel;
+            dstPtrDepth = dstPtrChannel;
+
+            for(int i = 0; i < roi.xyzwhdROI.roiDepth; i++)
+            {
+                Rpp32f *srcPtrRow, *dstPtrRow;
+                srcPtrRow = srcPtrDepth;
+                dstPtrRow = dstPtrDepth;
+
+                for(int j = 0; j < roi.xyzwhdROI.roiHeight; j++)
+                {
+                    Rpp32f *srcPtrTemp, *dstPtrTemp;
+                    srcPtrTemp = srcPtrRow;
+                    dstPtrTemp = dstPtrRow;
+
+                    int vectorLoopCount = 0;
+                    for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                    {
+#if __AVX2__
+                        __m256 p[1];
+
+                        rpp_simd_load(rpp_load8_f32_to_f32_avx, srcPtrTemp, p);    // simd loads
+                        compute_fmadd_8_host(p, pFmaddParams);                     // fmadd adjustment
+                        rpp_simd_store(rpp_store8_f32_to_f32_avx, dstPtrTemp, p);  // simd stores
+#endif
+                        srcPtrTemp += vectorIncrementPerChannel;
+                        dstPtrTemp += vectorIncrementPerChannel;
+                    }
+                    for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                    {
+                        *dstPtrTemp = (*srcPtrTemp * mulParam) + addParam;
+
+                        srcPtrTemp++;
+                        dstPtrTemp++;
+                    }
+
+                    srcPtrRow += srcGenericDescPtr->strides[3];
```

Comment:
srcGenericDescPtr->strides[3] won't always be the H stride to jump from row to row. True for NCDHW but not for NDHWC. For NDHWC, strides[2] will jump from row to row. Pls check.

Reply:
Fixed this.